clients/src/test/java/org/apache/kafka/clients/admin/NewTopicTest.java

```diff
@@ -0,0 +1,149 @@
+        assertEquals(TEST_TOPIC, topic.name());
+        assertEquals(NUM_PARTITIONS, topic.numPartitions());
+        assertEquals(REPLICATION_FACTOR, topic.replicationFactor());
+        assertNull(topic.replicasAssignments());
+    }
+
+    @Test
+    public void testConstructorWithOptionalValues() {
+        Optional<Integer> numPartitions = Optional.empty();
+        Optional<Short> replicationFactor = Optional.empty();
+        NewTopic topic = new NewTopic(TEST_TOPIC, numPartitions, replicationFactor);
+        assertEquals(TEST_TOPIC, topic.name());
+        assertEquals(NO_NUM_PARTITIONS, topic.numPartitions());
+        assertEquals(NO_REPLICATION_FACTOR, topic.replicationFactor());
+        assertNull(topic.replicasAssignments());
+    }
+
+    @Test
+    public void testConstructorWithReplicasAssignments() {
+        Map<Integer, List<Integer>> replicasAssignments = new HashMap<>();
+        replicasAssignments.put(0, BROKER_IDS);
+        NewTopic newTopic = new NewTopic(TEST_TOPIC, replicasAssignments);
+        assertEquals(TEST_TOPIC, newTopic.name());
+        assertEquals(NO_NUM_PARTITIONS, newTopic.numPartitions());
+        assertEquals(NO_REPLICATION_FACTOR, newTopic.replicationFactor());
+        assertEquals(replicasAssignments, newTopic.replicasAssignments());
+    }
+
+    @Test
+    public void testConfigs() {
+        NewTopic newTopic = new NewTopic(TEST_TOPIC, NUM_PARTITIONS, (short) REPLICATION_FACTOR);
+        Map<String, String> configs = new HashMap<>();
+        configs.put(CLEANUP_POLICY_CONFIG_KEY, CLEANUP_POLICY_CONFIG_VALUE);
+        newTopic.configs(configs);
+        assertEquals(configs, newTopic.configs());
+    }
+
+    @Test
+    public void testUnmodifiableReplicasAssignments() {
+        Map<Integer, List<Integer>> replicasAssignments = new HashMap<>();
+        replicasAssignments.put(0, BROKER_IDS);
+        NewTopic newTopic = new NewTopic(TEST_TOPIC, replicasAssignments);
+        Map<Integer, List<Integer>> returnedAssignments = newTopic.replicasAssignments();
+
+        assertThrows(UnsupportedOperationException.class, () -> {
+            returnedAssignments.put(1, Arrays.asList(3, 4));
+        });
+    }
+
+    @Test
+    public void testConvertToCreatableTopic() {
+        int partitionIndex = 0;
+        Map<Integer, List<Integer>> replicasAssignments = new HashMap<>();
+        replicasAssignments.put(partitionIndex, BROKER_IDS);
+        NewTopic topic = new NewTopic(TEST_TOPIC, replicasAssignments);
+        Map<String, String> configs = new HashMap<>();
+        configs.put(CLEANUP_POLICY_CONFIG_KEY, CLEANUP_POLICY_CONFIG_VALUE);
+        topic.configs(configs);
+
+        CreateTopicsRequestData.CreatableTopic creatableTopic = topic.convertToCreatableTopic();
+
+        assertEquals(TEST_TOPIC, creatableTopic.name());
+        assertEquals(NO_NUM_PARTITIONS, creatableTopic.numPartitions());
```

Comment:
Could you please add test case that `CreateTopicsRequestData.CreatableTopic#numPartitions` and `CreateTopicsRequestData.CreatableTopic#replicationFactor` return positive value?

Reply:
Ah yes, added one.